src/herbivoresAndCarnivores.js

```diff
@@ -1,15 +1,41 @@
 'use strict';
 
 class Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    this.name = name;
+    this.health = health;
+
+    if (Animal.alive === undefined) {
+      Animal.alive = [];
+    };
+    Animal.alive.push(this);
+  }
 }
 
 class Herbivore extends Animal {
-  // write your code here
+  constructor(name, health) {
+    super(name, health);
+    this.hidden = false;
+  }
+  hide() {
+    this.hidden = true;
+  }
 }
 
 class Carnivore extends Animal {
-  // write your code here
+  constructor(name, health = 100) {
+    super(name, health);
+  }
+
+  bite(bitten) {
+    if (bitten.hidden === false) {
+      bitten.health -= 50;
+    }
+
+    if (bitten.health <= 0) {
+      Animal.alive.splice(Animal.alive.indexOf(x => x === bitten), 1);
```

Comment:
Make use of **.filter** array method here
```suggestion
      Animal.alive = Animal.alive.filter(animal => animal.health > 0);
```

Checklist: _[CODE PERFOMANCE] - Try to avoid nesting loops (indexOf + splice), use filter instead_